backend/src/packages/users/user.service.ts

```diff
@@ -20,10 +20,24 @@ class UserService implements IService {
     return this.userRepository.find();
   }
 
-  public findByPhone(value: string): Promise<UserEntity | null> {
-    return this.userRepository.findByPhone(value);
+  public async findByPhoneAndEmail({
+    phone,
+    email,
+  }: UserSignUpRequestDto): Promise<
+    (UserEntityT & { createdAt: Date; updatedAt: Date }) | null
+  > {
+    const user = await this.userRepository.findByPhoneAndEmail({
+      phone,
+      email,
+    });
+
+    return user ?? null;
```

Comment:
I think u can remove ?? because repo call returns entity or null 
```suggestion
    return user;
```